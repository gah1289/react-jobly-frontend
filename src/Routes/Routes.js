import JoblyApi from '../api';
import axios from 'axios';
import { Routes, Route } from 'react-router-dom';
import React, { useContext } from 'react';
import Home from '../Home/Home';
import CompaniesList from '../Companies/CompaniesList';
import Company from '../Companies/Company';
import JobsList from '../Jobs/JobsList';

import Login from '../Users/Login';
import Signup from '../Users/Signup';
import Profile from '../Users/Profile';
import NotFound from '../NotFound';
import Logout from '../Users/Logout';

function JoblyRoutes() {
	return (
		<Routes>
			<Route path="/" element={<Home />} />
			<Route path="/companies" element={<CompaniesList />} />
			<Route path="/companies/:id" element={<Company />} />
			<Route path="/jobs" element={<JobsList />} />

			<Route path="/login" element={<Login />} />
			<Route path="/signup" element={<Signup />} />
			<Route path="/profile" element={<Profile />} />
			<Route path="/logout" element={<Logout />} />
			<Route path="*" element={<NotFound />} />
		</Routes>
	);
}

export default JoblyRoutes;
