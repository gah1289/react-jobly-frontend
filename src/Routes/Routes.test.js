import React from 'react';
import { render } from '@testing-library/react';
import Routes from './Routes';
import { MemoryRouter } from 'react-router';
import ItemContext from '../ItemContext';

describe('Renders Routes', function() {
	let isLoggedIn = true;

	let isLoading = false;

	let user = {
		username     : 'testUser',
		firstName    : 'Test',
		lastName     : 'User',
		email        : 'test@user.com',
		password     : 'password',
		isAdmin      : false,
		applications : []
	};

	user.current = user;
	it('renders without crashing', function() {
		render(
			<MemoryRouter>
				<ItemContext.Provider value={{ isLoading, isLoggedIn, user }}>
					<Routes />
				</ItemContext.Provider>
			</MemoryRouter>
		);
	});

	it('matches snapshot', function() {
		const { asFragment } = render(
			<MemoryRouter>
				<ItemContext.Provider value={{ isLoading, isLoggedIn, user }}>
					<Routes />
				</ItemContext.Provider>
			</MemoryRouter>
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
