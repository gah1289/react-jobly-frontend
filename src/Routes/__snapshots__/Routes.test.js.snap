// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Routes matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Hello, Test!
    </h1>
  </div>
</DocumentFragment>
`;
